websock/http/server.nim

```diff
@@ -191,22 +192,30 @@ proc accept*(server: HttpServer): Future[HttpRequest]
 
   trace "Got new request", isTls = server.secure
   try:
-    return await server.parseRequest(stream)
+    let
+      parseFut = server.parseRequest(stream)
+      # Failsafe at timeout * 1.05
+      timeout = server.handshakeTimeout + server.handshakeTimeout div 20
```

Comment:
I'm not sure this is the right approach, why would we have two timeouts one for the initial request in `parseRequest` another here, does it make sense to wait twice, whats the reasoning? I would really just stick with one timeout on `parseRequest().wait()` (or similar), wether it took too long to send the headers or during validation is irrelevant, IMO.

Reply:
I see, ok that makes sense then. In that case I would separate the two timeouts so that we can tweak them separately. For (contrived) example, we might know that a particular implementation is slow sending headers and adjust for that. We can either have them as two separate options - `headersTimeout` and `handshakeTimeout`, or amortize the `headersTimeout` into the `handshakeTimeout`, i.e. `handshakeTimeout + handshakeTimeout`.

The reason is because otherwise you're making arbitrary choices - why `handshakeTimeout + handshakeTimeout / 20` and not `handshakeTimeout + handshakeTimeout / 50`, this makes it difficult for the users/consumers to reason about it. Giving the user the ability to tweak this options makes the usage of the library more predictable and removes the need to for us to make this decisions for the user.